src/Tizen.NUI.Scene3D/src/public/Light/Light.cs

```diff
@@ -0,0 +1,121 @@
+
+namespace Tizen.NUI.Scene3D
+{
+    /// <summary>
+    /// </summary>
+    /// <remarks>
+    /// </remarks>
+    /// <example>
+    /// <code>
+    /// </code>
+    /// </example>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class Light : View
+    {
+        internal Light(global::System.IntPtr cPtr, bool cMemoryOwn) : base(cPtr, cMemoryOwn)
+        {
+        }
+
+        /// <summary>
+        /// Create an initialized Light.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Light() : this(Interop.Light.LightNew(), true)
+        {
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Copy constructor.
+        /// </summary>
+        /// <param name="light">Source object to copy.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Light(Light light) : this(Interop.Light.NewLight(Light.getCPtr(light)), true)
+        {
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+        }
+
+        /// <summary>
+        /// Assignment operator.
+        /// </summary>
+        /// <param name="light">Source object to be assigned.</param>
+        /// <returns>Reference to this.</returns>
+        internal Light Assign(Light light)
+        {
+            Light ret = new Light(Interop.Light.LightAssign(SwigCPtr, Light.getCPtr(light)), false);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
+        }
+
+        /// <summary>
+        /// Maximum Activated Light Count.
+        /// It is possible to add more Lights to the SceneView than the Maximum Activated Light Count,
+        /// but only up to the Maximum Activated Light Count Lights will work.
+        /// </summary>
+        /// <remarks>
+        /// This property is read only.
+        /// </remarks>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public uint MaximumActivatedLightCount
+        {
+            get
+            {
+                return Interop.Light.GetMaximumEnabledLightCount();
+            }
+        }
+
+        /// <summary>
+        /// The Light is turned on when the Light object is added on SceneView and enabled.
+        /// </summary>
+        /// <remarks>
+        /// SceneView can turn on only up to maximum enabled light count that can be retrieved by GetMaximumEnabledLightCount().
+        /// </remarks>
+        /// <param name="enable">True to make this Light enable.</param>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Enable(bool enable)
```

Comment:
In many cases, a C# class often provide a `IsEnabled` property instead of method to setting enabled value.
For example,
```C#
public bool IsEnabled
{
    get => Interop.Light.IsEnabled(SwigCPtr);
    set => Interop.Light.Enable(SwigCPtr, value);
}
```
How about combine `Enable()` and `IsEnabled()` into a property `IsEnabled` like the example?

Reply:
In our case, Light inherits View. And View already has IsEnabled property.
So, in the new patch, we changed the name to EnableLight and IsEnabledLight.

However I think this not the best way.
How do you think How we can fix it?

Reply:
IsLightEnabled property?
or 
current EnableLight and IsEnabledLight methods?
or
new other solution..?

Reply:
IMHO the class name itself is `Light`, so the `Light.IsLightEnabled` feels a bit redundant.

By the way, if the word `enable` here means that the on/off state of the light, what is the difference with belows?
(They are inherited from the `View`)
* Light.Visibility
* Light.Show(), Light.Hide()
* Light.Color = Color.Transparent;

Reply:
Yes Enable of light is to turn on/off.
However...
Light is not a bulb. So Light has not concept of visibility.
In NUI/DALi, Visibility, Show, and Hide are used to define the View will be rendered or not.
But, the Light is not renderable object. And it is just a conceptual item to change other object's visual result.
And also Light is not renderable object it's color is not includes transparency.
To be honest, to use new property of LightColor is more precise, but because we need to use View to control transform so we don't want to make the redundant too.
And, if we set Color to Black (same as 24 bit transparent), it looks same result with IsLightEnable is false.
But for the case App need to store original color value to re-turn on the light. and it looks unhappy.
We can simply imagine real world's bulb color and turn on/off switch.


So........ Which naming is better?ㅠㅠㅠㅠㅠ

Reply:
It seems that the `View` does not perfect fit to be a base class of the `Light` but I guess it would have been your best option in the current NUI struture since there is no class to express transformable. 
Anyway, if the `IsLightEnabled` is the name from DALi, it would be the best :)

Reply:
Agree. :)
`IsLightEnabled` looks fine.

Reply:
Thank you I changed it to the property.